components/board/genericlinux/data.go

```diff
@@ -282,10 +279,16 @@ func getBoardMapping(pinDefs []PinDefinition, gpioChipsInfo map[string]gpioChipD
 	for _, pinDef := range pinDefs {
 		key := pinDef.PinNumberBoard
 
-		gpioChipInfo, ok := gpioChipsInfo[pinDef.GPIOChipSysFSDir]
+		var ngpio int
+		for n := range pinDef.GPIOChipRelativeIDs {
+			ngpio = n
+			break // each gpio pin should only be associated with one gpiochip in the config
+		}
+
+		gpioChipInfo, ok := gpioChipsInfo[ngpio]
 		if !ok {
-			return nil, fmt.Errorf("unknown GPIO device %s for pin %d",
-				pinDef.GPIOChipSysFSDir, key)
+			return nil, fmt.Errorf("unknown GPIO device for chip with ngpio %d",
+				ngpio)
```

Comment:
Keep the `key` in the error, too: that will help people track down where the problem is (especially for large boards with many pins; it'll help narrow down which line to look at).